cmd/crypto/vault.go

```diff
@@ -0,0 +1,149 @@
+package crypto
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+
+	vault "github.com/hashicorp/vault/api"
+	"github.com/minio/minio/cmd/logger"
+)
+
+const (
+	// VaultEndpointEnv Vault endpoint environment variable
+	VaultEndpointEnv = "MINIO_VAULT_ENDPOINT"
+	// VaultAppRoleIDEnv  Vault App Role ID environment variable
+	VaultAppRoleIDEnv = "MINIO_VAULT_APP_ROLE_ID"
+	// VaultAppSecretIDEnv Vault App Secret ID environment variable
+	VaultAppSecretIDEnv = "MINIO_VAULT_APP_SECRET_ID"
+	// VaultAppVersionEnv Vault App Version environment variable
+	VaultAppVersionEnv = "MINIO_VAULT_APP_VERSION"
+	// VaultKeyNameEnv Vault Encryption Key Name environment variable
+	VaultKeyNameEnv = "MINIO_VAULT_KEYNAME"
+)
+
+type VaultService struct {
```

Comment:
This can be private. No need to export. Server should only work with `KMS` interface.